client/my-sites/media-library/google-photos-picker-button.tsx

```diff
@@ -22,11 +24,18 @@ const GooglePhotosPickerButton = () => {
 	};
 
 	useEffect( () => {
-		! session && createSession();
-	}, [ session, createSession ] );
+		const isSessionExpired =
+			session?.expireTime && moment( session.expireTime ).isBefore( new Date() );
+
+		if ( ! session || isSessionExpired ) {
```

Comment:
Not sure if this represents all the cases, should we also consider that session might be deleted in some cases and the session is not valid anymore?